docs/CONNECTOR_PROTOCOL.md

```diff
@@ -297,7 +301,7 @@ In addition to the connector index `.elastic-connectors`, we have an additional
 - `in_progress` -> A job is successfully started.
 - `canceling` -> The cancelation of the job is initiated.
 - `canceled` -> A job is canceled.
-- `suspended` -> A job is successfully started.
+- `suspended` -> A job has been suspended and will attempt to continue execution later.
```

Comment:
Yeah, 'suspended' could mean 'will never be picked up again' depending on circumstances.

Reply:
> it can also be initiated by customers.

I'm afraid I don't understand. We have "Canceled" for user-initiated cancels, who can initiate suspend?

> Yeah, 'suspended' could mean 'will never be picked up again' depending on circumstances.

What does 'suspended' actually means? If it can have meaning based on circumstances, should we split it into different statuses?

Reply:
I don't think so. Suspended means exactly what it means: the job has been suspended. What happens with that job afterwards depends on implementation and context, but the fact that multiple things can happen doesn't represent a different job status.

Reply:
> Suspended means exactly what it means: the job has been suspended

I can't understand what it means - it's ambiguous. What does it pragmatically means?

What statuses can suspended job transition to?
In context of task of resuming jobs, can this status be used to resume jobs or should I create another?

Reply:
I don't see the ambiguity? 'Suspended' is the state it's in right now. It means at one point it was picked up, and then work on it stopped before it was finished without it erroring or being cancelled. That's the state of the job. In general, we should try to resume a job that's in that state. But we can never guarantee that it will be resumed.

The point I'm trying to make is that we cannot know for certain ahead of time whether a suspended job will be picked up in the future, because all sort of things can happen. There are situations where you don't want to resume a job (eg: when a newer job was picked up and completed afterwards, making this one obsolete). There are situations where you want to resume a job but can't (eg: the data source has been deleted). And there are many (probably most) situations in which you can resume the job and should. 

We don't need different states to model this, because we cannot model "this is suspended but the next time the job will be retried the data source has been deleted". Or "this is suspended and when we try to pick it up again another job will have been completed so this one is now obsolete". These are not states we can predict, and we shouldn't try to. 

Essentially, jobs once started can stop in different ways:
- They can error out, in which case they won't be resumed
- They can be canceled, in which case they won't be resumed
- They can be suspended, in which case we may try to pick them up again, depending on the circumstances when we try to do so

If we try to pick up a suspended job and can't, for whatever reason, we should probably transition it to a different state (assuming we don't want to retry again later on). Which state depends on context.

Reply:
That change is fine, @vidok's change of 'may' is also fine. 

In general you should probably use suspended when the connector crashes or is interrupted, unless we expect the type of crash to be caused by an issue that will cause future resumptions to also crash.

Reply:
I see, thanks again for the explanation!

Reply:
late to the discussion.

`suspended` was introduced when unifying statuses between crawler and connector. It only applies to crawler now and we may apply it to connector in the future:

1. a running crawl request is marked as `suspended` when enterprise search is shut down.
2. when enterprise search starts again, it will try to pick up crawl requests in `pending` (start a new crawl) and `suspended` (resume a suspended crawl) status.
3. There's no option to manually resume a crawl request, as there's no option to manually suspend a crawl request as well.